main.tf

```
@@ -0,0 +1,111 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source = "hashicorp/azurerm"
+      version = ">= 3.8"
+    }
+  }
+  backend "azurerm" {
+    resource_group_name  = "Cohort28_RajSal_ProjectExercise"
+    storage_account_name = "rsm12st01"
+    container_name       = "rsm12st01container"
+    key                  = "terraform.tfstate"
+    }
+}
+
+provider "azurerm" {
+  features {}
+}
+
+data "azurerm_resource_group" "main" {
+  name     = "Cohort28_RajSal_ProjectExercise"
+}
+
+# AppServicePlan
+resource "azurerm_service_plan" "main" {
+  name                = "terraformed-asp"
+  location            = data.azurerm_resource_group.main.location
+  resource_group_name = data.azurerm_resource_group.main.name
+  os_type             = "Linux"
+  sku_name            = "B1"
+}
+
+# AppService
+resource "azurerm_linux_web_app" "main" {
+  name                = "rs-m12-app"
+  location            = data.azurerm_resource_group.main.location
+  resource_group_name = data.azurerm_resource_group.main.name
+  service_plan_id     = azurerm_service_plan.main.id
+
+  site_config {
+    application_stack {
+      docker_image     = "rsalhan/m8-todo-app"
+      docker_image_tag = "prod"
+    }
+  }
+
+  app_settings = {
+    # API_KEY, BASE_URL, BOARD_ID, DOING_IDLIST, DONE_IDLIST, TODO_IDLIST, TOKEN
+    "DOCKER_REGISTRY_SERVER_URL" = "https://docker.io"
+    "FLASK_APP" = "todo_app/app."
```

I think we don't typically expect the `.` at the end here, although it may be forgiven (as `app.main` for example would refer to the main variable/function in the app file)
```suggestion
    "FLASK_APP" = "todo_app/app"
```

---

Thanks, my mistake, that's now been corrected.